Refactor and Add vagrant support

b4ldr opened the pull request:
Hi,

This pull request is quite large so appolagise for that but i got a bit carried away.  The bulk of the change it to move the config parameters around so they can be set when calling the module directly e.g.

``` puppet
class {'::icinga':
  icinga_user => nagios,
}
```

and so that settings can be set in multiple parts for the yaml e.g.

``` yaml
icinga::icinga_user: nagios
icinga::gui::gui_type: web
```

This moves away from setting everything under icinga::params:: and from my experience seems to be more in line with common practice.  The other refactoring i did was to move templates so the reflect the location they will be installed to on the bases system which seems much more intuative to me.  

I have also added a number of features

```
$manage_users:          only run icinga::users class if true
$manage_dbs:             if true and {web/ido}_db_server is mysql then create the mysql db's
$manage_repo:            only install third party repos id true
$web_auth_user_file:   location of the htpasswd user auth file
$web_auth_group_file: location of the htpasswd group auth file
$web_auth_users:        A hash of basic auth users to create
icinga::gui::basic_user: custom type to create basic auth users
```

Finally i have created a Vagrant file to test the module, I have readjusted some of the ordering and the vagrant box now comes up cleanly without errors

b4ldr committed:
refactor and add vagrant

rendhalver commented:
I had planned on moving the variables from params into the base class next.
I haven't had the time to get that done so thanks for that.
I will have a look through this when I get a chance.

Thanks

b4ldr committed:
Merge pull request #1 from icann-dns/v1.2.2-rc11-ICANN

refactor and add vagrant

b4ldr committed:
fix auth.xml.erb typo

b4ldr committed:
Merge pull request #2 from icann-dns/v1.2.2-rc11-ICANN

fix auth.xml.erb typo

rendhalver commented:
Sorry for the delay. I will try and have a look at this this evening.

b4ldr commented:
No worries its a big change so didn't expect it to go quickly

rendhalver commented on manifests/config.pp:
```diff
@@ -1,35 +1,33 @@
+  $stalking             = $::icinga::stalking,
+  $flap_detection       = $::icinga::flap_detection,
+) {
 
```
Is there a reason you have put these in as class params?
I see these classes as private so they shouldn't have any parameters.
Their values should come in from the icinga class as you have here but without being parameters
Same would apply to the rest of the internal subclasses.

b4ldr replied:
Sorry i missed your comments.  This is more just habit but i do some times use it to fine tune, however i dont think i  override anything at this level so would be happy for these to just be variable definitions at the top of the file

rendhalver commented on manifests/config.pp:
```diff
@@ -45,46 +43,37 @@
-    require => Class[icinga::install],
-  }
-
   file { '/etc/icinga/icinga.cfg':
```
There may have been a reason I was managing this but I can't think of it now.
It's best we don't if we are supporting $::osfamily == RedHat as well. :)

rendhalver commented on manifests/gui.pp:
```diff
@@ -96,88 +129,85 @@
+    $ssl_cert   = "${apache::ssl_path}/${webhostname}.crt"
+    $ssl_key    = "${apache::ssl_path}/${webhostname}.key"
+    $ssl_ca     = $ssl_cacrt
+    $ssl_cipher = $ssl_cypher_list
```
Good plan. :)

rendhalver commented on manifests/gui/basic_user.pp:
```diff
@@ -0,0 +1,14 @@
+  $web_auth_user_file  = $::icinga::web_auth_user_file
+  exec {"create gui::basic_user ${name}":
+    user    => $::apache::user,
+    command => "/usr/bin/htpasswd -b ${web_auth_user_file} ${name} ${password}",
```
Sneaky. :)
Thanks for this.

rendhalver commented on manifests/init.pp:
```diff
@@ -3,30 +3,274 @@
+    include icinga::users
+    Class['icinga::users'] -> Class['icinga::install']
+  }
+  if $manage_dbs {
```
I tend to leave database management out of my modules and leave it as something people can manage with their profiles.
Everyone does it different (I use mariadb and galera with exported resources) so it's not worth the trouble of attempting to support one use case.

b4ldr replied:
Im also happy with that

rendhalver commented on manifests/gui.pp:
```diff
@@ -2,65 +2,98 @@
-    $ido_db_port = 5432
-  } else {
-    $ido_db_port = $icinga::params::ido_db_port
-  }
```
I was doing this so people could use a different port for their database server and set a reasonable default.

b4ldr replied:
Yes i got the logic i just moved it to the erb file, personal preference just think it looks cleaner there and

rendhalver commented on manifests/install.pp:
```diff
@@ -2,84 +2,75 @@
 
   case $::osfamily {
     'RedHat' : {
-      if ($::operatingsystem != 'Fedora') {
```
Does Fedora packages have this package now?

rendhalver commented on templates/etc/icinga/resource.cfg.erb:
```diff
@@ -5,7 +5,7 @@
 $USER1$=<%= @nagios_plugins %>
 
-<% if @db_password != '' -%>
+<% if @db_password -%>
```
We will need to test this works as expected in the puppet versions we support.

rendhalver commented on tests/Puppetfile:
```diff
@@ -0,0 +1,2 @@
+forge 'https://forge.puppetlabs.com'
+mod 'icann-icinga', :path => '/vagrant'
```
Typo? :)

rendhalver commented:
I finally had some time to look at this.
See comments :)

b4ldr commented:
Thanks for the comments and sorry for the delay, ill fix up the few things you mentioned over the next few days

rendhalver commented:
Thanks for the patch. :)
I understand getting busy and not having spare time.

b4ldr committed:
Remove the abillity to manage the database an example of how to
configure the database is used by the vagrant install.

Make parameters to most manifests private.  I have left the parameters
for gui.pp and idoconfig.pp avalible so that one can fine tune them

b4ldr commented:
Ok i have updated the code to remove the manage_db cappability.  I have also made the majority of parameters private, however i have left them configurable for gui.pp and idoconfig.pp as this seemed somewhat intuitive.

rendhalver commented:
Awesome!

Thanks for that. I shall get that looked at as soon as I get a chance.
Then we can merge it and push out a new release. :)

rendhalver commented:
Sorry I forgot about this one.
I will merge this today.
I have another issue that needs fixing as well to I will do my best to get that sorted out as well and then make a new release.
